denonavr/denonavr.py

```diff
@@ -419,6 +433,26 @@ def send_post_command(self, command, body):
                 "Host %s returned HTTP status code %s to POST command at "
                 "end point %s"), self._host, res.status_code, command)
 
+    def get_device_info(self):
+        """Get device information."""
+        url = "http://{host}:{port}{command}".format(
+            host=self._host, port=DESCRIPTION_URL[self._receiver_type].port,
+            command=DESCRIPTION_URL[self._receiver_type].url)
+        
+        device_info = evaluate_scpd_xml(url)
+        
+        if device_info == False:
+            self._manufacturer = "Denon"
+            self._model_name = "Unknown"
+            self._serial_number = self._host
```

Comment:
I think the IP address shouldn't be used in Home Assistant as `unique_id` because it can changed. The MAC address will be a better choice.

Reply:
I do not use the IP address, only if getting the serialNumber fails I use the IP address as a backup, but I also print an error in the log about it, so users that experiance problems getting the serialNumber will see that error and hopefully report it such that it can be fixed. The IP adress is then only used in the meantime for these users to be able to still setup the denonavr platform.

I could also just simply remove this "alternative" behaviour...

I do not know the MAC address of the receiver. It is specified in the device_info.xml, but I do not have a way of retrieving it from the appcommand so on newer avr-x-2016 models I have no way of getting the MAC, therefore implementing MAC at this point would break it for the avr-x-2016 models.

Unless someone could figure out how to get the MAC form the appcommand?

Reply:
But the `serial_number` property shouldn't contain the IP address, even if the real serial number cannot be obtained. Maybe the library should return `None` when there is no serial number and the HA integration should get the MAC address via `getmac` package?

Reply:
I removed the ip adress being used as serial_number, I will also make a change to the homeassistant PR